src/operation/overlayng/OverlayPoints.cpp

```diff
@@ -27,6 +27,47 @@ namespace geos {      // geos
 namespace operation { // geos.operation
 namespace overlayng { // geos.operation.overlayng
 
+struct PointExtractingFilter: public GeometryComponentFilter {
+
+    PointExtractingFilter(std::map<Coordinate, std::unique_ptr<Point>>& p_ptMap, const PrecisionModel* p_pm)
+        : ptMap(p_ptMap), pm(p_pm)
+    {}
+
+    void
+    filter_ro(const Geometry* geom)
+    {
+        const Point* pt = dynamic_cast<const Point*>(geom);
```

Comment:
Faster to use `geom->getGeometryTypeId()`, if that matters.

Reply:
Good to use the best-practice pattern - I'll change this.

Reply:
And then `static_cast<const Point*>(geom)`?

Reply:
Yes